### DISCUSSION TITLE
Local Redirect Issue

### DISCUSSION THREAD

**Discussion by billyors** [lostb1t/replex#39] [Q&A] [answered]
Big fan of the project so far!

My external clients coming in to watch Plex remotely are working great. Here's the arch:

Remote Client --> Custom URL (https://replex.custom.domain) --> CF Proxy --> Pfsense --> Traefik --> Internal Replex Host (192.168.8.8:3001) --> Proxies to Plex for streaming (192.168.8.8:32400)

This config is working great as-is from the outside. But, the internal clients on the same network as Plex are struggling to ever hit Replex (LAN network cidr is added to settings). I have an internal DNS record so that replex.custom.domain resolves locally. The internal Plex clients are skipping right past Replex now. What am I missing here?

It feels like dns caching on the clients (Roku, Web, Android, HTPC). As I can _sometimes_ see Replex on first startup of a client, then after a refresh Replex collections/heroes vanish and we're directly connecting to Plex. Any assistance is much appreciated 🙏

**Comment by lostb1t** [comment]
I'm not sure as I don't use it internally. But I know plex is bad in managing the different routings. It's registery usually hold old entries. Might want to check what endpoints are still avaiable internally. for example plex has docker endpoints that it tries to connect to. This is my instance. It can't connect to it but if it could it might pick that above the domain. might wanna check it out for your server with something like https://apps.apple.com/nl/app/varys-for-plex/id1450380518?l=en-GB (used for the screenshot) there are other options like inspect the network panel in the web app.

I'm the end the trick is to block everything out to plex except traffic coming from your reverse proxy. 

![image](https://github.com/sarendsen/replex/assets/168401/1d7244f0-16ef-444a-9b7e-ada03d46d9a8)

**Reply by billyors** [reply]
Thanks for the input. I'll continue digging around to see which endpoint(s) is still open for connection outside of replex. Will update here if I find a solution.

**Reply by lostb1t** [reply, marked as answer]
Ow en try to disable gdm in the network: Enable local network discovery (GDM

Also if running docker for plex don't use the host network option. As it will bind to your server ip. Just open the 32400 port with docker

**Reply by billyors** [reply]
Just stepped away, will test and update later. I am doing both of those things. GDM enabled and host networking for plex. I had a hunch one of these might be the culprit.

**Reply by billyors** [reply]
Strange results. Disabling GDM sped up discovery, so that's cool.

I also updated my plex container to use the default networking (rather than host). It looks like things are working! Mostly.

Externally, Replex still looks great. Internally, my Web, Android and HTPC clients all look great. However, Roku specifically is managing to still somehow connect to Plex directly. I looked at the debug logs from the Roku side and could see it discovering both replex and the Plex 32400 server. I have no idea how it's the only one that can still see it. All others seem great though. I'll keep digging on this. Thanks again for the assist!

**Reply by billyors** [reply]
Can confirm it was DNS caching on the roku devices. All looks good after the ttl expired. Thank you!
